Make dev dependencies compatible with PHP 8

omarlopesino opened the pull request:
After applying this pull request the project won't have any composer restrictions to use PHP 8. It will let us check if the project is compatible with PHP 8.

omarlopesino committed:
DEV - Require dev dependencies compatible with PHP 8

omarlopesino committed:
GRUMPHP - Compatibility with latest grumphp version

omarlopesino committed:
Readd git commit msg

omarlopesino committed:
GRUMPHP - Readd md and js extensions

omarlopesino committed:
GRUMPHP - Revert change

dlopez-akalam commented on composer.json:
```diff
@@ -56,14 +56,13 @@
+        "metadrop/grumphp-drupal-check": "^0.3.0",
         "metadrop/scripthor": "1.x-dev",
-        "mglaman/drupal-check": "^1.1",
+        "mglaman/drupal-check": "^1.1.8",
```
IMO it would be better to keep mglaman/drupal-check and phpro/grumphp dependencies on metadrop/grumphp package for better maintainability.

omarlopesino replied:
Ok for mglaman/drupal-check, but I think we need grumphp even we don't use metadrop/grumphp-drupal-check. I will remove only mglaman/drupal-check

dlopez-akalam commented on grumphp.yml:
```diff
@@ -1,13 +1,8 @@
-  git_dir: .
-  bin_dir: vendor/bin
+  hooks_dir: ~
+  hooks_preset: local
```
In the documentation says `Presets are only used when you did NOT specify a custom hooks_dir`
https://github.com/phpro/grumphp/blob/master/doc/parameters.md

Is this line necessary?

omarlopesino replied:
Not really, that's true. Plus, local is the default value. Fixing that.

dlopez-akalam commented on grumphp.yml:
```diff
@@ -1,13 +1,8 @@
-parameters:
-  git_hook_variables:
-    EXEC_GRUMPHP_COMMAND: docker-compose exec -T php
```
Why are we removing the docker support by default?

omarlopesino replied:
This was accidentally removed, my bad. Fixing it.

omarlopesino committed:
GRUMPHP - Remove unneded configuration

hooks_preset has already local at is default value. Plus, it is not
neccesary for us.

omarlopesino committed:
GRUMPHP - Add back the configuration to run checks inside PHP docker container

omarlopesino committed:
DRUPAL CHECK - Remove drupal check as we rely in grumphp for that

omarlopesino commented:
@dlopez-akalam I've solved the issues you've found. May you check again?

dlopez-akalam commented on composer.json:
```diff
@@ -56,14 +56,12 @@
         "phpmd/phpmd": "^2.9",
-        "sebastian/phpcpd": "^5.0",
-        "vijaycs85/drupal-quality-checker": "^1.3.0"
+        "phpro/grumphp": "^1.3.0"
```
I would remove this dependency because is already declared on phpro/grumphp
https://github.com/Metadrop/grumphp-drupal-check/blob/master/composer.json

omarlopesino replied:
I think I didn't submit my last comment where I explained why I don't want remove it.

I understand that if grumphp is indirectly required so we could remove it, reducing maintenance. 

However, we need grumphp in our project with or without the grumphp drupal check  plugin. If we need to use grumphp for more reasons than using grumphp-drupal-check (phpcs, phpcpd, git commit msg), then why should we define grumphp as an indirect dependency. I think it should be direct dependency.

omarlopesino replied:
I've been clarified by David in a call, in the last commit I explain the reason explaining why we drop the dependency.

omarlopesino committed:
GRUMPHP - Remove grumphp as direct dependency

We rely in the version added in grumphp-drupal-check. In the case we
remove grumphp-drupal-check, we'll add phpro/grumphp again
